packages/mdl-drawer/README.md

```diff
@@ -0,0 +1,284 @@
+
+```css
+.content {
+  display: flex;
+  height: 100%;
+  box-sizing: border-box;
+}
+
+main {
+  display: inline-block;
+  padding: 16px;
+  box-sizing: border-box;
+}
+```
+
+```html
+<body>
+  <div>Toolbar goes here</div>
+
+  <div class="content">
+    <nav class="mdl-permanent-drawer mdl-typography">
+      <nav id="icon-with-text-demo" class="mdl-list">
+        <a class="mdl-list-item mdl-permanent-drawer--selected" href="#">
+          <i class="material-icons mdl-list-item__start-detail" aria-hidden="true">inbox</i>Inbox
+        </a>
+        <a class="mdl-list-item" href="#">
+          <i class="material-icons mdl-list-item__start-detail" aria-hidden="true">star</i>Star
+        </a>
+      </nav>
+    <main>
+      Page content goes here.
+    </main>
+  </div>
+</body>
+```
+
+CSS classes:
+
+| Class                                  | Description                                                                |
+| -------------------------------------- | -------------------------------------------------------------------------- |
+| `mdl-permanent-drawer`                 | Mandatory. Needs to be set on the root element of the component.           |
+| `mdl-permanent-drawer__content`        | Mandatory. Needs to be set on the container node for the drawer content.   |
+| `mdl-permanent-drawer__toolbar-spacer` | Optional. Add to node to provide the matching amount of space for toolbar. |
+
+
+## Temporary drawer usage
+
+A temporary drawer is usually closed, sliding out at a higher elevation than the content when opened. It is appropriate
+for any display size.
+
+```html
+<aside class="mdl-temporary-drawer mdl-typography">
+  <nav class="mdl-temporary-drawer__drawer">
+    <header class="mdl-temporary-drawer__header">
+      <div class="mdl-temporary-drawer__header-content">
+        Header here
+      </div>
+    </header>
+    <nav id="icon-with-text-demo" class="mdl-list">
+      <a class="mdl-list-item mdl-temporary-drawer--selected" href="#">
+        <i class="material-icons mdl-list-item__start-detail" aria-hidden="true">inbox</i>Inbox
+      </a>
+      <a class="mdl-list-item" href="#">
+        <i class="material-icons mdl-list-item__start-detail" aria-hidden="true">star</i>Star
+      </a>
+      </a>
+    </nav>
+  </nav>
+</aside>
+```
+
+```js
+let drawer = new mdl.TemporaryDrawer(document.querySelector('.mdl-temporary-drawer'));
+document.querySelector('.menu').addEventListener('click', () => drawer.open = true);
+```
+
+### Headers and toolbar spacers
+
+Temporary drawers can use toolbar spacers, headers, or neither.
+
+A toolbar spacer adds to the drawer the same amount of space that the toolbar takes up in your application. This is
+very useful for visual alignment and consistency. Note that you can place content inside the toolbar spacer.
+
+```html
+<aside class="mdl-temporary-drawer mdl-typography">
+  <nav class="mdl-temporary-drawer__drawer">
+
+    <div class="mdl-permanent-drawer__toolbar-spacer"></div>
```

Comment:
it may confuse devs to have a `mdl-permanent-drawer` element inside of a `mdl-temporary-drawer` block. What if we renamed this element to `mdl-drawer-toolbar-spacer` similar to `mdl-list-divider`?